### ISSUE TITLE
Grouped Labels on X-axis for Stacked Bar charts

### ISSUE THREAD

**Issue by albert-kp** [recharts/recharts#3562]
- [ ] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
The stacked bar chart component should allow the grouped labels to be shown in x-axis.
![fc3noor2](https://user-images.githubusercontent.com/117899068/235667734-97bd600f-6183-420b-993b-e4c332881541.png)

### What does the proposed API look like?
 groups: [
              { title: '2019', cols: 4 },
              { title: '2020', cols: 4 }
            ]

The Xaxis component should accept a groups prop which should accept the group title and number of cols to span

**Comment by nikolasrieble** [comment]
It would be great if we could do this in an easier way. In the meantime, as a workaround, do consider to simply use multiple xAxes.

**Comment by albert-kp** [comment]
> It would be great if we could do this in an easier way. In the meantime, as a workaround, do consider to simply use multiple xAxes.

Yes. Currently acheived the same using multiple xAxes as you have mentioned. But it would be great if we could have a functionality build around X-axis component.

**Comment by nikolasrieble** [comment]
If you have achieved this, would you like to share your workaround with others, i.e. by adding an example to storybook?

**Comment by jdavidferreira** [comment]
See this: https://stackoverflow.com/questions/66202932/recharts-customized-x-axis

**Comment by sepruda** [comment]
I have multiple x-axis, the problem is that it's impossible to center the labels on the second x-axis to align with the first x-axis like i want. The issue I have [here](https://github.com/recharts/recharts/issues/3598) is closed
